app/scripts/platforms/extension.js

```diff
@@ -45,37 +45,10 @@ export default class ExtensionPlatform {
   }
 
   getVersion() {
-    const { version, version_name: versionName } =
-      browser.runtime.getManifest();
-
-    const versionParts = version.split('.');
-    if (versionName) {
-      if (versionParts.length < 4) {
-        throw new Error(`Version missing build number: '${version}'`);
-      }
-      // On Chrome, a more descriptive representation of the version is stored in the
-      // `version_name` field for display purposes. We use this field instead of the `version`
-      // field on Chrome for non-main builds (i.e. Flask, Beta) because we want to show the
-      // version in the SemVer-compliant format "v[major].[minor].[patch]-[build-type].[build-number]",
-      // yet Chrome does not allow letters in the `version` field.
-      return versionName;
-      // A fourth version part is sometimes present for "rollback" Chrome builds
-    } else if (![3, 4].includes(versionParts.length)) {
-      throw new Error(`Invalid version: ${version}`);
-    } else if (versionParts[2].match(/[^\d]/u)) {
-      // On Firefox, the build type and build version are in the third part of the version.
-      const [major, minor, patchAndPrerelease] = versionParts;
-      const matches = patchAndPrerelease.match(/^(\d+)([A-Za-z]+)(\d)+$/u);
-      if (matches === null) {
-        throw new Error(`Version contains invalid prerelease: ${version}`);
-      }
-      const [, patch, buildType, buildVersion] = matches;
-      return `${major}.${minor}.${patch}-${buildType}.${buildVersion}`;
-    }
-
-    // If there is no `version_name` and there are only 3 or 4 version parts, then this is not a
-    // prerelease and the version requires no modification.
-    return version;
+    // METAMASK_VERSION is a semver compliant version number, e.g., `1.2.3`
+    // for pre-release builds it includes the build type and pre-release number.
+    // e.g., '1.2.3-beta.4'
+    return process.env.METAMASK_VERSION;
```

Comment:
Hmm. So this might not be enough, depending on what we want here.

You're right that this should match what `getVersion` returned previously in most cases. But the bundle of code running might be from a different version of the application than the manifest. This isn't supposed to happen (to my knowledge), but we've seen it before in Sentry.

But that difference is really only relevant for diagnostic logs/state snapshots, which accounts for.... two of the callsites for this function? For the rest, `process.env.METAMASK_VERSION` is just as good. And for the two diagnostic state snapshot cases, we don't really need the SerVer translation of the version, the raw version would be enough to tip us off that something weird was happening.

My suggestion would be to:

* Update this function to return `browser.runtime.getManifest().version` directly, no translation
* Replace `platform.getVersion` calls with `process.env.METAMASK_VERSION` everywhere except in `getCleanAppState` and `globalThis.stateHooks.getSentryState`. Those are the only two places where I can imagine the raw manifest version being helpful.